vyper/builtin_functions/functions.py

```diff
@@ -857,21 +903,38 @@ class Extract32(_SimpleBuiltinFunction):
 
     _id = "extract32"
     _inputs = [("b", BytesArrayPrimitive()), ("start", SignedIntegerAbstractType())]
+    # "name_literal" is a placeholder value for TypeTypeDefinition
     _kwargs = {"output_type": Optional("name_literal", "bytes32")}
     _return_type = None
 
     def fetch_call_return(self, node):
         super().fetch_call_return(node)
+        return_type = self.infer_kwarg_types(node)["output_type"].typedef
+        return return_type
+
+    def infer_arg_types(self, node):
+        validate_call_args(node, 2, list(self._kwargs.keys()))
+        validate_expected_type(node.args[0], BytesArrayDefinition())
+        b_type = get_possible_types_from_node(node.args[0]).pop()
+        validate_expected_type(node.args[1], Int128Definition())
+
+        return [b_type, Int128Definition()]
+
+    def infer_kwarg_types(self, node):
         if node.keywords:
-            return_type = get_type_from_annotation(node.keywords[0].value, DataLocation.MEMORY)
+            output_type = get_type_from_annotation(node.keywords[0].value, DataLocation.MEMORY)
             if not isinstance(
-                return_type, (AddressDefinition, Bytes32Definition, IntegerAbstractType)
+                output_type, (AddressDefinition, Bytes32Definition, IntegerAbstractType)
             ):
-                raise
+                raise InvalidType(
+                    "Output type must be one of bytes32, int128 or address", node.keywords[0].value
```

Comment:
error message does not match behavior

Reply:
I noticed the docs specify `int128`. Should it be updated to all integer types as well?

Reply:
To update docs to `uint256` and remove negative integer test cases